src/main/java/com/thevoxelbox/voxelsniper/util/LocationWrapper.java

```diff
@@ -0,0 +1,50 @@
+package com.thevoxelbox.voxelsniper.util;
+
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.data.BlockData;
+
+import javax.annotation.Nullable;
+import java.util.Objects;
+
+public class LocationWrapper extends Location {
+
+    public LocationWrapper(World world, double x, double y, double z) {
+        super(world, x, y, z);
+    }
+    public LocationWrapper(Location l) {
+        super(l.getWorld(), l.getX(), l.getY(), l.getZ());
+    }
+
+    public void setBlockData(BlockData blockData, boolean b) {
+        getClampedBlock().setBlockData(blockData, b);
+    }
+
+    public Block getClampedBlock() {
+        return Objects.requireNonNull(getWorld()).getBlockAt(getBlockX(), getYClamped(), getBlockZ());
+    }
+
+    public Block getOffsetBlock(int x, int y, int z) {
+        return Objects.requireNonNull(getWorld()).getBlockAt(getBlockX() + x, getBlockY() + y, getBlockZ() + z);
+    }
+
+    public Block getOffsetClampedBlock(int x, int y, int z) {
+        return Objects.requireNonNull(getWorld()).getBlockAt(getBlockX() + x, getYClamped(y), getBlockZ() + z);
+    }
```

Comment:
```suggestion
    public Block getOffsetBlock(int x, int y, int z) {
        return Objects.requireNonNull(getWorld()).getBlockAt(getBlockX() + x, getYClamped(y), getBlockZ() + z);
    }
```

I think we should always clamp Y, and not include a method that allows us to bypass this.

Reply:
I disagree. If you were to make a tree at y=315 you wouldn't want the leaves to be clamped since that would place them all at y=319. Expected behaviour would be for it to just be cut off. Though after testing this seems to not be the case

Reply:
I am not sure what happens when we use getBlockAt() out of range?

Reply:
Going tot test that since this seems to not work correctly right now

Reply:
Will come back to the clamping later